variant_remapping_tools/reads_to_remapped_variants.py

```diff
@@ -7,75 +7,66 @@
-    for operator in range(0, flank_length + 2 + local_region_size + cig_list[0:flank_length+1].count(pysam.CDEL)):
-        # Stop incrementing varpos once we've reached read_var_pos:
-        if start + operator < read_var_pos:
-            # Mismatch/matches and deletions increase the position counter:
-            if cig_list[operator] == pysam.CMATCH or cig_list[operator] == pysam.CDEL:
-                varpos += 1
-        # If we are in the local region around the variant, but not at the position of the variant:
-        if ((start + operator >= (read_var_pos - local_region_size))
-                and (start + operator <= (read_var_pos + local_region_size))
-                and (start + operator != read_var_pos)):
-            if cig_list[operator] == 0:  # Match or mismatch
-                perf_counter += 1  # Increase the counter for perfect local region
-    return varpos, perf_counter, local_region_size
-
-
-def is_read_valid(read, counter, flank_length, score_cutoff, diff_cutoff):
-    """
-    Filter out reads that are:
-        - Unmapped
-        - Non primary (secondary): AS's that are too low and XS's that are too close to AS
-    """
-    if read.is_secondary:  # We only want to deal with primary reads
-        return False
-    counter['total'] += 1
-    if read.is_unmapped:  # Can be decoded with bitwise flag with & 4 (4 means unmapped)
-        counter['unmapped'] += 1
-        return False
-    # Getting AS and XS:
-    AS = read.get_tag('AS')
-    try:
-        XS = read.get_tag('XS')  # Some variants don't have secondary alignments, which throws an error
-    except KeyError:
-        XS = -flank_length  # Set an arbitrary low value for the 'artificial' secondary alignment
-    if AS <= int(score_cutoff):
-        counter['primary_poor'] += 1
-        return False
-    if AS - XS < diff_cutoff:
-        counter['gap_small'] += 1
-        return False
-    return True
-
-
-def is_local_alignment_perfect(counter, perf_counter, local_region_size):
+def reverse_complement(sequence):
+    return str(Seq(sequence, generic_dna).reverse_complement())
+
+
+def calculate_new_variant_definition(left_read, right_read, ref_fasta):
     """
-    Filter out non-perfect local alignment around the variant
+    Resolve the variant definition from the flanking region alignment and old variant definition
+    TODO: Link to algorithm description once public
     """
-    if perf_counter != 2 * local_region_size:
-        counter['context_bad'] += 1
-        return False
-    # Filter out AS's that are too low and XS's that are too close to AS, and those with a non-perfect local alignment:
-    counter['remapped'] += 1
-    return True
+    # grab information from the read name
+    name = left_read.query_name
+    info = name.split('|')
+    old_ref = info[2]
+    old_alts = info[3].split(',')
+
+    # Define new ref and new pos
+    new_ref = fetch_bases(ref_fasta, left_read.reference_name, left_read.reference_end + 1,
```

Comment:
I think we also need to assert that `left_read.reference_name` and `right_read.reference_name` are the same. In rare cases, they may be different either due to aligner error, or because of genuine genome reassembly

Reply:
This should be covered by the `pass_basic_filtering` function which will filter out any read pair where the reference name are different.